dali/imgcodec/decoders/nvjpeg/memory_pool.cc

```diff
@@ -0,0 +1,295 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cassert>
+#include <fstream>
+#include <iostream>
+#include <memory>
+#include <shared_mutex>
+#include <utility>
+#include <unordered_map>
+#include "dali/imgcodec/decoders/nvjpeg/memory_pool.h"
+#include "dali/core/cuda_error.h"
+#include "dali/core/mm/malloc_resource.h"
+#include "dali/pipeline/data/buffer.h"
+
+namespace dali {
+
+namespace imgcodec {
+
+namespace nvjpeg_memory {
```

Comment:
If this doesn't depend on nvjpeg anymore, I'd move it out of this dir, and remove any references to `nvjpeg` from the names, to make it clear:
```
decoders/memory_pool.*
decoders/nvjpeg/nvjpeg_memory.*
decoders/nvjpeg/nvjpeg2k_memory.*
```